evalml/model_understanding/graphs.py

```diff
@@ -608,21 +608,30 @@ def partial_dependence(pipeline, X, features, percentiles=(0.05, 0.95), grid_res
     """
 
     X = infer_feature_types(X)
-    # Dynamically set the grid resolution to the maximum number of categories
-    # in the categorical variables if there are more categories than resolution cells
+    # Dynamically set the grid resolution to the maximum number of values
+    # in the categorical/datetime variables if there are more categories/datetime values than resolution cells
     X_cats = X.select("categorical")
-    if X_cats.shape[1] != 0:
+    is_categorical = [_is_feature_of_type(features, X, ww.logical_types.Categorical)] if not isinstance(features, (list, tuple)) else [_is_feature_of_type(f, X, ww.logical_types.Categorical) for f in features]
+    if X_cats.shape[1] != 0 and any(is_categorical):
         max_num_cats = max(X_cats.describe().loc["nunique"])
```

Comment:
I think rather than looking at the max over all categoricals, we need to look at the max of the categorical features passed in by the user. My concern is that if there's a dataset with a numeric feature, a categorical feature with 10 values and a categorical feature with 100 values, and the user wants to compute the partial dependence of the numeric column and the column with 10 values, the grid size will be set to 100. That may slow down the entire process, especially if the user sets a 10 <= grid_size < =  100.

Let me know what you think. I think it's in scope of #2116.

Reply:
yup thats a good catch - I'll make the changes!